packages/frenetic/frenetic.5.0.5/opam

```diff
@@ -0,0 +1,48 @@
+opam-version: "2.0"
+synopsis: "The Frenetic Programming Language and Runtime System"
+maintainer: "Nate Foster <jnfoster@cs.cornell.edu>"
+authors: "Arjun Guha <arjun@cs.umass.edu>, Nate Foster <jnfoster@cs.cornell.edu>, Steffen Smolka <smolka@cs.cornell.edu>"
+homepage: "http://frenetic-lang.org"
+license: "LGPL-3.0-or-later"
+bug-reports: "https://github.com/frenetic-lang/frenetic/issues"
+dev-repo: "git+https://github.com/frenetic-lang/frenetic.git"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["dune" "build" "@doc" "-p" name "-j" jobs] {with-doc}
+]
+depends: [
+  "ocaml" {build & >= "4.11.0"}
+  "dune" {build & >= "2.0"}
+  "odoc"
+  #########################
+  "async"  {>= "v0.14.0" }
+  "base64" {>= "3.5.0"}
+  "cohttp" {>= "4.0.0"}
+  "cohttp-async" {>= "4.0.0" }
+  "core"   {>= "v0.14.0" }
+  "cstruct" {>= "6.0.1"}
+  "cstruct-sexp" {>= "6.0.1"}
+  "ipaddr" {>= "5.2.0"}
+  "menhir" {>= "20210419"}
+  "mparser" {>= "1.3"}
+  "mparser-re" {>= "1.3"}
+  "ocamlgraph" {>= "2.0.0"}
+  "open" {>= "0.2.1"}
+  "ppxlib" {>= "0.22.0"}
+  "ppx_jane" {>= "0.14.0"}
+  "ppx_cstruct" {>= "5.2.0"}
+  "ppx_deriving" {>= "5.1"}
+  "sedlex" {>= "2.2" & <= "2.3"}
```

Comment:
```suggestion
  "sedlex" {>= "2.2" & < "2.4"}
```
Using a strict equality sign is usually more correct (e.g. a 2.3.1 is always a possibility)

Also, just curious, what is the reason behind this restriction?

Reply:
I had the same error as FStar here: https://github.com/FStarLang/FStar/issues/2343

Reply:
cc @toots ?

Reply:
Thanks for letting me know! As fas as I can tell, `sedlex` `2.4` has a clean `mli` interface. This issue is due to an embedded mli of the module shipped with the projet that prevents it from linking with the new API: https://github.com/frenetic-lang/frenetic/blob/master/src/lib/netkat/Lexer.cppo.ml#L13

I believe that the missing functionalities for tracking source locations were added into Sedlex a while ago. I'd suggest to either update to a recent version or rename the custom module.

Reply:
Ok. I did the same hack as FStar, adding a `__private__next_int` field to our `LexBuffer` overlay... it's ugly, but at least we avoid brittle version pinning for now.